v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/models/PreparedStatementValueObject.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates.models;
+
+public class PreparedStatementValueObject<T> {
+  private String dataType;
+  private T value;
```

Comment:
Do we have a strong reason to make a copy of `T`?

The context on this:
Some of the customer workloads we have got  in past (and have had production issues) were to do with customers having rows with of the order of megabyte values - Imagine binary blobs of few Mbs in size.

In such cases making a copy of `T` could increase the chance of out of memory errors.

One thing we have done on the [reader-path](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/d9243b5392d659df6ada2d947a43a088c56a15f1/v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/cassandra/rowmapper/CassandraFieldMapper.java) is to only map elements without caching them anywhere. We followed the same for `JDBC` as well.